flexboxAPIspecV2.md

```diff
@@ -0,0 +1,783 @@
+
+# API Details
+
+
+
+```c# (but really MIDL3)
+
+runtimeclass FlexboxLayout;
+
+    enum FlexboxAlignContent
+        {
+            Start = 0,
+            End = 1,
+            Center = 2,
+            Stretch = 3,
+            SpaceBetween = 4,
+            SpaceAround = 5,
+            SpaceEvenly = 6
+        }
+```
+
+
+
+## Flex Order
+
+The order property specifies the order of a flexible item relative to the rest of the flexible items inside the same container.
+
+Note: If the element is not a flexible item, the order property has no effect.
+
+
+```xml
+        <controls:LayoutPanel>
+            <controls:LayoutPanel.Layout>
+                <controls:FlexboxLayout/>
+            </controls:LayoutPanel.Layout>
+            <Grid controls:FlexboxLayout.Order="5">
+                <TextBlock>5</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Order="-1">
+                <TextBlock>-1</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Order="2">
+                <TextBlock>2</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Order="1">
+                <TextBlock>1</TextBlock>
+            </Grid>
+        </controls:LayoutPanel>
+```
+
+### Output of the above code (styling omitted)
+![image](./FlexOrder.PNG)
+
+
+##  Flex Order Accepted Values
+
+
+
+| Value | Description |
+|-|-|
+| Number (Default Value = 0) | Specifies the order for the flexible item |
+
+# API Details
+
+
+```c# (but really MIDL3)
+
+unsealed runtimeclass FlexboxLayout : NonVirtualizingLayout
+    
+    {
+        static Windows.UI.Xaml.DependencyProperty OrderProperty{ get; };
+        static void SetOrder(Windows.UI.Xaml.UIElement element, Int32 value);
+        static Int32 GetOrder(Windows.UI.Xaml.UIElement element);
+    }
+```
+
+
+
+## Flex Grow
+
+The flex-grow property specifies how much the item will grow relative to the rest of the flexible items inside the same container.
+
+Note: If the element is not a flexible item, the flex-grow property has no effect.
+
+
+```xml
+ <controls:LayoutPanel MinWidth="500">
+            <controls:LayoutPanel.Layout>
+                <controls:FlexboxLayout/>
+            </controls:LayoutPanel.Layout>
+            <Grid controls:FlexboxLayout.Grow="0" controls:FlexboxLayout.Basis="100" >
+                <TextBlock>0</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Grow="1" controls:FlexboxLayout.Basis="100">
+                <TextBlock>1</TextBlock>
+            </Grid>
+            <Grid controls:FlexboxLayout.Grow="0" controls:FlexboxLayout.Basis="100">
+                <TextBlock>0</TextBlock>
+            </Grid>
+        </controls:LayoutPanel>
+```
+
+### Output of the above code (styling omitted)
+![image](./FlexGrow.PNG)
+
+
+##  Flex Grow Accepted Values
+
+
+
+| Value | Description |
+|-|-|
+| Number (Default Value = 0) | Specifies how much the item will grow relative to the rest of the flexible items. |
```

Comment:
this (and others) aren't numbers, but a struct comprised of a number and a `FlexboxUnit`.
Need to define this struct in this doc too.

Reply:
Could you give me an example of what this might look like?

Reply:
You would first define something like:
```cs
struct FlexboxMeasurement
{
    Double Value;
    FlexboxUnit Unit;
}
```

then you would use that type in here, so the user can specify "20%" and that gets converted to a `FlexboxMeasurement` (name pending) with Value=20, and Unit=Percentage